container-search/src/main/java/com/yahoo/prelude/query/CompositeItem.java

```diff
@@ -219,7 +219,6 @@ public CompositeItem clone() {
         copy.subitems = new java.util.ArrayList<>();
         for (Item subItem : subitems) {
             Item subItemCopy = subItem.clone();
-            copy.adding(subItemCopy);
             copy.subitems.add(subItemCopy);
```

Comment:
I think you need to do subItemCopy.setParent(copy) then.

Reply:
Done